### PR TITLE
update upgrade guide

### PR DESCRIPTION
**Pull request by megan07**
updating upgrade doc

```release-note:breaking-change
`access_context_manager`: Made `os_type` required on block `google_access_context_manager_access_level.basic.conditions.device_policy.os_constraints`.
```

```release-note:breaking-change
`app_engine`: Made `script_path` required on `google_app_engine_standard_app_version.handlers.script`.
```

```release-note:breaking-change
`bigquery`: Made `role` required on `google_bigquery_dataset.access`.
```

```release-note:breaking-change
`bigtable`: Made exactly one of `single_cluster_routing` or `multi_cluster_routing_use_any` required on `google_bigtable_app_profile`.
```

```release-note:breaking-change
`binary_authorization`: Made `name_pattern` required on `google_binary_authorization_policy.admission_whitelist_patterns`.
```

```release-note:breaking-change
`binary_authorization`: Made `evaluation_mode` and `enforcement_mode` required on `google_binary_authorization_policy.cluster_admission_rules`.
```

```release-note:breaking-change
`cloudbuild`: Made exactly one of `filename` or `build` required on `google_cloudbuild_trigger`.
```

```release-note:breaking-change
`cloudscheduler`: Made exactly one of `pubsub_target`, `http_target` or `app_engine_http_target` required on `google_cloudscheduler_job`.
```

```release-note:breaking-change
`compute`: Made exactly one of `http_health_check`, `https_health_check`, `http2_health_check`, `tcp_health_check` or `ssl_health_check` required on `google_compute_health_check`.
```

```release-note:breaking-change
`compute`: Made exactly one of `http_health_check`, `https_health_check`, `http2_health_check`, `tcp_health_check` or `ssl_health_check` required on `google_compute_region_health_check`.
```

```release-note:breaking-change
`monitoring`: Made exactly one of `http_check` or `tcp_check` is now required on `google_monitoring_uptime_check_config`.
```

```release-note:breaking-change
`monitoring`: Made at least one of `auth_info`, `port`, `headers`, `path`, `use_ssl`, or `mask_headers` is now required on `google_monitoring_uptime_check_config.http_check`
```

### CODE REVIEW

File: third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown [hunk 1]
```diff
@@ -236,17 +261,236 @@ resource "google_cloudiot_registry" "myregistry" {
 }
 ```
 
+### `logging_service` and `monitoring_service` defaults changed
```
Comment (paddycarver) [inline third_party/terraform/website/docs/guides/version_3_upgrade.html.markdown]: I think this needs to get moved to container_cluster after the rebase.

File: products/cloudfunctions/api.yaml [hunk 2]
```diff
@@ -136,17 +136,30 @@ objects:
         description: |
           The Google Cloud Storage URL, starting with gs://, pointing to the zip
           archive which contains the function.
+        exactly_one_of:
```
Comment (megan07) [inline products/cloudfunctions/api.yaml]: These changes actually don't do anything, because this resource is not currently generated.  I removed the update from the upgrade guide, but I figured I could leave them all in here so that when we eventually do generate it, they'll be there.

File: products/cloudrun/api.yaml [hunk 3]
```diff
@@ -272,7 +272,6 @@ objects:
     properties:
     - !ruby/object:Api::Type::NestedObject
       name: template
-      required: true
```
Comment (paddycarver) [inline products/cloudrun/api.yaml]: Is this change intentional? Seems out of character with the rest of the changes.

Comment (megan07) [reply products/cloudrun/api.yaml]: Yes!  I took that out because i saw its all flattened in terraform.yaml, and the flattened part is already required.
